HeterogeneousCore/CUDAServices/bin/cudaIsEnabled.cpp

```diff
@@ -0,0 +1,31 @@
+#include <algorithm>
+#include <array>
+#include <cstdlib>
+#include <iostream>
+
+#include <cuda_runtime.h>
+
+int main() {
+  int devices = 0;
+  auto status = cudaGetDeviceCount(& devices);
+  if (status != cudaSuccess) {
+    return EXIT_FAILURE;
+  }
+
+  int minimumMajor = 6; // min minor is implicitly 0
+
+  // This approach (requiring all devices are supported) is rather
+  // conservative. In principle we could consider just dropping the
+  // unsupported devices. Currently that would be easiest to achieve
+  // in CUDAService though.
```

Comment:
In fact, as the only requirement seems to be "at least one device with capability above *something*`, wouldn't it make more sense to change the check below to
```c++
    ...

    if (properties.major >= minimumMajor) {
      return EXIT_SUCCESS;
    }
  }

  return EXIT_SUCCESS;
}
```
?

Reply:
I'm wondering if the minimum capability should be a property of the module rather than of the process.

Reply:
(for a later PR)

Reply:
Not for this or any approach we ever discussed: IIUC, this check is done once per job when parsing the configuration, to decide whether or not to enable CUDA at all.

Also, we compile all modules for all supported architectures; there is no concept if minimal sm architecture at the module level.

Reply:
In fact, a nice improvement could be to set the supported architectures based on the scram cuda flags.

Reply:
> In fact, as the only requirement seems to be "at least one device with capability above _something_`, wouldn't it make more sense to change the check below to
> 
> ```c++
>     ...
> 
>     if (properties.major >= minimumMajor) {
>       return EXIT_SUCCESS;
>     }
>   }
> 
>   return EXIT_SUCCESS;
> }
> ```
> 
> ?

Currently the requirement is to require that *all* devices have capability above something. Moving to  "at least one above" would require also communicating to `CUDAService` which devices are used and which are not. I'd leave that to later time.

Reply:
Created an issue #280.